Traceback (most recent call last):
  File "/opt/anaconda3/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/opt/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 1112, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/anaconda3/lib/python3.8/site-packages/nbclient/util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/opt/anaconda3/lib/python3.8/site-packages/nbclient/util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "/opt/anaconda3/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/opt/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 553, in async_execute
    await self.async_execute_cell(
  File "/opt/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 857, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/opt/anaconda3/lib/python3.8/site-packages/nbclient/client.py", line 760, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import numpy as np
import numpy.random as npr
import numpy_financial as npf
import pandas as pd
%matplotlib inline
from pylab import mpl, plt
import seaborn as sns
mpl.style.use('seaborn')
mpl.rcParams['figure.figsize'] = (10, 6)

import\
    datetime as dt,\
    json,\
    requests,\
    time
import warnings
warnings.filterwarnings('ignore')

import panel as pn
pn.extension('plotly')
import plotly.express as px
import hvplot.pandas

from dotenv import find_dotenv, get_key
alpaca_api_key = get_key(find_dotenv(), 'ALPACA_API_KEY')
alpaca_secret_key = get_key(find_dotenv(), 'ALPACA_SECRET_KEY')
import alpaca_trade_api as tradeapi
from alpaca_trade_api.rest import REST, TimeFrame
api = REST(
    alpaca_api_key,
    alpaca_secret_key,
    api_version='v2',
)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mValueError[0m                                Traceback (most recent call last)
[0;32m<ipython-input-1-6d411625e795>[0m in [0;36m<module>[0;34m[0m
[1;32m     27[0m [0;32mimport[0m [0malpaca_trade_api[0m [0;32mas[0m [0mtradeapi[0m[0;34m[0m[0;34m[0m[0m
[1;32m     28[0m [0;32mfrom[0m [0malpaca_trade_api[0m[0;34m.[0m[0mrest[0m [0;32mimport[0m [0mREST[0m[0;34m,[0m [0mTimeFrame[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 29[0;31m api = REST(
[0m[1;32m     30[0m     [0malpaca_api_key[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m
[1;32m     31[0m     [0malpaca_secret_key[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m

[0;32m/opt/anaconda3/lib/python3.8/site-packages/alpaca_trade_api/rest.py[0m in [0;36m__init__[0;34m(self, key_id, secret_key, base_url, api_version, oauth, raw_data)[0m
[1;32m     90[0m                          [0mEntity[0m [0mobjects[0m[0;34m.[0m[0;34m[0m[0;34m[0m[0m
[1;32m     91[0m         """
[0;32m---> 92[0;31m         self._key_id, self._secret_key, self._oauth = get_credentials(
[0m[1;32m     93[0m             key_id, secret_key, oauth)
[1;32m     94[0m         [0mself[0m[0;34m.[0m[0m_base_url[0m[0;34m:[0m [0mURL[0m [0;34m=[0m [0mURL[0m[0;34m([0m[0mbase_url[0m [0;32mor[0m [0mget_base_url[0m[0;34m([0m[0;34m)[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;32m/opt/anaconda3/lib/python3.8/site-packages/alpaca_trade_api/common.py[0m in [0;36mget_credentials[0;34m(key_id, secret_key, oauth)[0m
[1;32m     80[0m     [0mkey_id[0m [0;34m=[0m [0mkey_id[0m [0;32mor[0m [0mos[0m[0;34m.[0m[0menviron[0m[0;34m.[0m[0mget[0m[0;34m([0m[0;34m'APCA_API_KEY_ID'[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m     81[0m     [0;32mif[0m [0mkey_id[0m [0;32mis[0m [0;32mNone[0m [0;32mand[0m [0moauth[0m [0;32mis[0m [0;32mNone[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 82[0;31m         raise ValueError('Key ID must be given to access Alpaca trade API',
[0m[1;32m     83[0m                          ' (env: APCA_API_KEY_ID)')
[1;32m     84[0m [0;34m[0m[0m

[0;31mValueError[0m: ('Key ID must be given to access Alpaca trade API', ' (env: APCA_API_KEY_ID)')
ValueError: ('Key ID must be given to access Alpaca trade API', ' (env: APCA_API_KEY_ID)')

